packages/api/src/controllers/taskController.js

```diff
@@ -295,11 +297,42 @@ const taskController = {
     };
   },
 
-  async checkAndAddCustomType(typeOfTask, data, farm_id) {
+  async checkCustomDependencies(typeOfTask, data, farm_id) {
+    const irrigationTypeValues = {
+      farm_id,
+      irrigation_type_name: data.irrigation_type?.irrigation_type_name,
+      default_measuring_type: data.irrigation_type?.default_measuring_type,
+    };
     switch (typeOfTask) {
       case 'field_work_task': {
         return await this.checkAndAddCustomFieldWork(data, farm_id);
       }
+      case 'irrigation_task':
+        if (data.irrigation_type?.irrigation_task_type_other) {
+          await IrrigationTypesModel.insertCustomIrrigationType({ ...irrigationTypeValues });
+        }
+        if (data.irrigation_type?.set_default_irrigation_task_type_measurement) {
+          await IrrigationTypesModel.query()
+            .where('irrigation_type_name', irrigationTypeValues.irrigation_type_name)
+            .patch({
+              default_measuring_type: irrigationTypeValues.default_measuring_type,
+              farm_id: irrigationTypeValues.farm_id,
+            })
+            .returning('*');
+        }
+        if (data.location_defaults) {
+          for (const location_default of data.location_defaults) {
+            await locationDefaultsModel.transaction(async (trx) => {
```

Comment:
can shift this line to LocationDefaultsModel file. can add this in the static function.